src/app/home/recent-locations/recent-locations.component.ts

```diff
@@ -55,14 +57,33 @@ export class RecentLocationsComponent implements OnInit {
     console.log(this.locations);
   }
 
+  updateSessionState() {
+    this.ngOnInit();
```

Comment:
updateSessionState would be called both times when user signs in or out. We should check what is the session state.